go/vt/vtgate/engine/plan_description.go

```diff
@@ -152,7 +152,11 @@ func PrimitiveDescriptionFromMap(data map[string]any) (pd PrimitiveDescription,
 		}
 	}
 	if ttt, isPresent := data["TargetTabletType"]; isPresent {
-		pd.TargetTabletType = topodatapb.TabletType(ttt.(int))
+		val, ok := topodatapb.TabletType_value[ttt.(string)]
+		if !ok {
+			panic(fmt.Sprintf("TargetTabletType is not a valid tablet type: %v", ttt))
+		}
+		pd.TargetTabletType = topodatapb.TabletType(val)
 	}
```

Comment:
what caused this change? 🤔

Reply:
the code before was just wrong. tablet_types are not encoded as ints, they are strings in the json, so what was there before was panicking trying to cast a string into an int